source/auth/auth.rst

```diff
@@ -683,6 +683,236 @@ mechanism
+
+==== ==================== ================= ============================================================================================================================================== 
+Name Friendly Name        Type              Description
+==== ==================== ================= ==============================================================================================================================================
+r    client nonce         BinData Subtype 0 32 byte cryptographically secure random number 
+p    gs2-cb-flag          int32             The integer representation of the ASCII charater 'n' or 'y', i.e., ``110`` or ``121``
+s    server nonce         BinData Subtype 0 64 bytes total, 32 bytes from the client first message and a 32 byte cryptographically secure random number generated by the server
+h    sts host             string            FQDN of the STS service 
+a    authorization header string            Authorization header for `AWS Signature Version 4 <https://docs.aws.amazon.com/general/latest/gr/signature-version-4.html?shortFooter=true>`_
+d    X-AMZ-Date           string            Current date in UTC. See `AWS Signature Version 4 <https://docs.aws.amazon.com/general/latest/gr/signature-version-4.html?shortFooter=true>`_
+t    X-AMZ-Security-Token string            Optional AWS security token
+==== ==================== ================= ============================================================================================================================================== 
+
+Drivers MUST NOT advertise support for channel binding, as the server does
+not support it and legacy servers may fail authentication if drivers advertise
+support. The client-first-message MUST set the gs2-cb-flag to the integer representation 
+of the ASCII character ``n``, i.e., ``110``.
+
+Conversation
+````````````
+As an example, given a client nonce value of "enJwWTtNSkR+WztFZCE3d1NWSiMpfU54YCgmPU5lY1Q=", a MONGODB-IAM conversation decoded from
+BSON to JSON would appear as follows:
+
+Client First
+
+.. code:: javascript
+
+   { 
+        "r" : new BinData(0, "dzw1U2IwSEtgaWI0IUxZMVJqc2xuQzNCcUxBc05wZjI="),
+        "p" : 110
+   }
+
+Server First
+
+.. code:: javascript
+
+   { 
+       "s" : new BinData(0, "dzw1U2IwSEtgaWI0IUxZMVJqc2xuQzNCcUxBc05wZjIGS0J9EgLwzEZ9dIzr/hnnK2mgd4D7F52t8g9yTC5cIA=="),
+       "h" : "sts.amazonaws.com"
+   }
+
+Client Second
+
+.. code:: javascript
+
+   { 
+       "a" : "AWS4-HMAC-SHA256 Credential=AKIAICGVLKOKZVY3X3DA/20191107/us-east-1/sts/aws4_request, SignedHeaders=content-length;content-type;host;x-amz-date;x-mongodb-gs2-cb-flag;x-mongodb-server-nonce, Signature=ab62ce1c75f19c4c8b918b2ed63b46512765ed9b8bb5d79b374ae83eeac11f55",
+       "d" : "20191107T002607Z"
+   }
+|
+Each message above will be encoded as BSON V1.1 objects and sent to the server as the value of ``payload``. Therefore, the SASL conversation would appear as:
+
+Client First
+
+.. code:: javascript
```

Comment:
Looking at the rendered version of these snippets at https://github.com/bazile-clyde/specifications/blob/SPEC-1451/source/auth/auth.rst#mongodb-iam, there are not appearing at all.

Reply:
Done